apps/io-wallet-user-func/src/infra/http/handlers/create-wallet-instance.ts

```diff
@@ -31,6 +40,30 @@ type WalletInstanceRequestPayload = t.TypeOf<
   typeof WalletInstanceRequestPayload
 >;
 
+// [SIW-1560] to do - a mock function that return the user email by the fiscal code
+const getUserEmailByFiscalCode = (
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  fiscalCode: string,
+): TE.TaskEither<Error, string> =>
+  pipe(TE.left(new Error("Not implemented yet")));
+
+export const sendEmailToUser: (
+  params: SendEmailNotificationParams,
+) => RTE.ReaderTaskEither<
+  { emailNotificationService: EmailNotificationService },
+  Error,
+  void
+> =
+  (params) =>
+  ({ emailNotificationService }) =>
+    pipe(
+      TE.tryCatch(
+        emailNotificationService.sendEmail(params),
+        (error) => new Error(`Error sending the mail to the user: ${error}`),
+      ),
+      TE.chain(() => TE.right(undefined)),
+    );
```

Comment:
`emailNotificationService.sendEmail(params)` it's already a taskeither.
```suggestion
    pipe(
      params,
      emailNotificationService.sendEmail
    );
```

Reply:
**Suggestion:** You should only use `TE.tryCatch` when you have a function that returns a `Promise` and does not use `fp-ts`.

Reply:
How cool!